// library/WaitDialog.js
//------------------------------------------------------------------------------
//
//     This code was auto generated.
//
//     Manual changes to this file may cause unexpected behavior in your application.
//     Manual changes to this file will be overwritten if the code is regenerated.
//
//------------------------------------------------------------------------------

const extend = require('js-base/core/extend');
const FlexLayout = extend(require('sf-core/ui/flexlayout'));
const ActivityIndicator = extend(require('sf-core/ui/activityindicator'));
const GifImageView = extend(require('sf-core/ui/gifimageview'));
const ImageView = extend(require('sf-core/ui/imageview'));

function addChild(childName, ChildClass, componentInstance) {
	this.children = this.children || {};
	this.children[childName] = new ChildClass(componentInstance);
	if (this.layout) this.layout.addChild(this.children[childName]);
	else this.addChild(this.children[childName]);
}

// Constructor
function WaitDialog(_super, props) {
	// Initalizes super class for this component scope

	_super(this, props);
	addChild.call(this, 'indicator', $WaitDialog$$Indicator_, this);
	addChild.call(this, 'gifIndicator', $WaitDialog$$GifIndicator_, this);
	addChild.call(this, 'imgIndicator', $WaitDialog$$ImgIndicator_, this);
}
WaitDialog.$$styleContext = {
	classNames: '.sf-flexLayout .waitDialog',
	defaultClassNames: '.default_common .default_flexLayout',
	userProps: {}
};
const WaitDialog_ = FlexLayout(WaitDialog);

function $WaitDialog$$Indicator(_super, pageInstance) {
	_super(this);

	pageInstance.indicator = this;
}
$WaitDialog$$Indicator.$$styleContext = {
	classNames: '.sf-activityIndicator .waitDialog-indicator',
	defaultClassNames: '.default_common .default_activityIndicator',
	userProps: {}
};
const $WaitDialog$$Indicator_ = ActivityIndicator($WaitDialog$$Indicator);

function $WaitDialog$$GifIndicator(_super, pageInstance) {
	_super(this);

	pageInstance.gifIndicator = this;
}
$WaitDialog$$GifIndicator.$$styleContext = {
	classNames: '.sf-gifImageView .waitDialog-gifIndicator',
	defaultClassNames: '.default_common .default_gifImageView',
	userProps: {}
};
const $WaitDialog$$GifIndicator_ = GifImageView($WaitDialog$$GifIndicator);

function $WaitDialog$$ImgIndicator(_super, pageInstance) {
	_super(this);

	pageInstance.imgIndicator = this;
}
$WaitDialog$$ImgIndicator.$$styleContext = {
	classNames: '.sf-imageView .waitDialog-imgIndicator',
	defaultClassNames: '.default_common .default_imageView',
	userProps: {}
};
const $WaitDialog$$ImgIndicator_ = ImageView($WaitDialog$$ImgIndicator);

module.exports = WaitDialog_;
